packages/react-union/src/scanning/core.js

```diff
@@ -0,0 +1,73 @@
+import o from 'ramda/src/o';
+import map from 'ramda/src/map';
+import find from 'ramda/src/find';
+import whereEq from 'ramda/src/whereEq';
+import ifElse from 'ramda/src/ifElse';
+import is from 'ramda/src/is';
+import prop from 'ramda/src/prop';
+import mergeDeepRight from 'ramda/src/mergeDeepRight';
+import compose from 'ramda/src/compose';
+
+import { validateDescriptorStructures, validateRoutesWithDescriptors } from '../validation';
+
+import {
+	getWidgetDescriptors as getCheerioWidgetDescriptors,
+	getCommonData as getCheerioCommonData,
+} from './descriptors/cheerio';
+
+import {
+	getWidgetDescriptors as getDomWidgetDescriptors,
+	getCommonData as getDomCommonData,
+} from './descriptors/dom';
+
+const getWidgetDescriptors = ifElse(
+	is(Function),
+	getCheerioWidgetDescriptors,
+	getDomWidgetDescriptors
+);
+
+const getCommonData = ifElse(is(Function), getCheerioCommonData, getDomCommonData);
+
+const mergeCommonDataToConfigs = commonData =>
+	map(mergeDeepRight({ descriptor: { data: commonData } }));
+
+const getConfigs = (routes, descriptors) => {
+	const findRouteByDescriptor = ({ widget }) => find(whereEq({ path: widget }), routes);
+	const findComponentByDescriptor = o(prop('component'), findRouteByDescriptor);
+
+	const findDescriptorConfig = descriptor => ({
+		component: findComponentByDescriptor(descriptor),
+		descriptor,
+	});
+
+	return map(findDescriptorConfig, descriptors);
+};
+
+const pairCommonDataWithConfigs = commonData => configs => ({ commonData, configs });
```

Comment:
I'd like to find a better name for this function. I'm using the name `scanResult` for the resulting structure, so maybe something like `getScanResult`. The function below can be renamed to `createScanResult` instead, as `get` implies that the function "does less" compared to `create`.